CA_DataUploaderLib/FilterVectorExpansion.cs

```diff
@@ -0,0 +1,89 @@
+﻿#nullable enable
+using CA_DataUploaderLib.Helpers;
+using CA_DataUploaderLib.IOconf;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CA_DataUploaderLib
+{
+    public class FilterVectorExpansion
+    {
+        private readonly List<IOconfFilter> _decisionFilters;
+        private readonly List<FilterSample> _legacyInputFilters;
+        private readonly CALogLevel _logLevel;
+
+        /// <remarks>
+        /// <paramref name="inputs"/> is updated by adding classic input filters and removing hidden sources.
+        /// </remarks>
+        public FilterVectorExpansion(List<VectorDescriptionItem> inputs, Func<IEnumerable<IOconfFilter>> getFilters, CALogLevel logLevel)
```

Comment:
would you rather inject the getFilters Func - instead of just the list of filters?